waveform_editor/options.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import os
+from pathlib import Path
+
+import panel as pn
+import param
+import yaml
+
+logger = logging.getLogger(__name__)
+
+_xdg = os.environ.get("XDG_CONFIG_HOME")
+_config_home = Path(_xdg) if _xdg else Path(os.environ["HOME"]) / ".config"
+CONFIG_FILE = _config_home / "waveform_editor.yaml"
+
+
+class NiceOptions(param.Parameterized):
+    executable = param.String(
+        label="NICE executable path", doc="Path to NICE inverse IMAS MUSCLE3 executable"
+    )
+    environment = param.Dict(
+        label="NICE environment variables", doc="Environment variables for NICE"
+    )
+    modules = param.List(label="NICE modules", doc="EasyBuild Modules to load for NICE")
+    md_pf_active = param.String(
+        label="'pf_active' machine description URI",
+    )
+    md_pf_passive = param.String(
+        label="'pf_passive' machine description URI",
+    )
+    md_wall = param.String(
+        label="'wall' machine description URI",
+    )
+    md_iron_core = param.String(
+        label="'iron_core' machine description URI",
+    )
+
+    def apply_settings(self, params):
+        """Update parameters from a dictionary."""
+        self.param.update(**params)
+
+    def to_dict(self):
+        """Returns a dictionary representation of current parameter values."""
+        return {p: getattr(self, p) for p in self.param if p != "name"}
+
+
+class UserConfig(param.Parameterized):
+    gs_solver = param.Selector(objects=["NICE"], default="NICE")
+
+    nice = param.ClassSelector(class_=NiceOptions, default=NiceOptions())
+
+    def __init__(self, **params):
+        super().__init__(**params)
+        self._load_settings()
+        self._save_settings()
+        self.param.watch(self._save_settings, list(self.param))
+        self.nice.param.watch(self._save_settings, list(self.nice.param))
+
+    def _load_settings(self):
+        """Load settings from disk and apply them to the current instance."""
+        if os.path.exists(CONFIG_FILE):
+            with open(CONFIG_FILE) as f:
+                settings = yaml.safe_load(f) or {}
+        else:
+            settings = {}
+
+        if "nice" in settings:
+            self.nice.apply_settings(settings["nice"])
+
+        base_settings = {k: v for k, v in settings.items() if k != "nice"}
+        self.param.update(**base_settings)
+
+    def _save_settings(self, event=None):
+        """Serialize current configuration to disk in YAML format."""
+        config = {
+            p: getattr(self, p) for p in self.param if p != "name" and p != "nice"
+        }
+
+        if self.gs_solver == "NICE":
+            config["nice"] = self.nice.to_dict()
+
+        with open(CONFIG_FILE, "w") as f:
```

Comment:
This will fail if the `_config_home` directory doesn't exist:

```python
>>> import pathlib
>>> open(pathlib.Path("/tmp/.config/test.yaml"), "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/.config/test.yaml'
```